ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfig.scala

```diff
@@ -16,16 +16,28 @@ case class WorkflowConfig(
 
 object WorkflowConfig {
 
-  case class SubmissionConfig(
+  sealed trait SubmissionConfig extends Product with Serializable {
+    def numberOfInstances: Int
+    def numberOfObservers: Int
+    def uniqueParties: Boolean
+  }
+
+  final case class FooSubmissionConfig(
```

Comment:
I'd rename this this back to `SubmissionConfig` and  `sealed trait SubmissionConfig` to `sealed trait SubmissionConfigBase` to have fewer changes in this PR.

We can revisit these names when we add another subclass of the base trait. And it'll be easier to judge the renames by contrasting different concrete flavours of the base config.

Reply:
I did a different trick, should be more history now.